### PR TITLE
wasi: fix closing of file descriptors

### PR DESCRIPTION
**Pull request by achille-roussel**
This PR addresses a bug that was causing the host fd 0 to be closed unexpectedly when any guest file descriptors were closed.

The bug was introduced by the refactor in #38

### CODE REVIEW

File: imports/builder_unix.go [hunk 1]
```diff
@@ -101,12 +99,22 @@ func (b *Builder) Instantiate(ctx context.Context, runtime wazero.Runtime) (ctxr
...
-		{stdin, "/dev/stdin"},
-		{stdout, "/dev/stdout"},
-		{stderr, "/dev/stderr"},
+		{stdin, syscall.O_RDONLY, "/dev/stdin"},
+		{stdout, syscall.O_WRONLY, "/dev/stdout"},
+		{stderr, syscall.O_WRONLY, "/dev/stderr"},
 	} {
+		var err error
+		if stdio.fd < 0 {
+			stdio.fd, err = syscall.Open(stdio.path, stdio.open, 0)
```
Comment (achille-roussel) [inline imports/builder_unix.go:112]: I changed this so the module never implicitly shares the underlying files used by the host fds.

File: internal/sockets/dial.go [hunk 2]
```diff
@@ -13,17 +13,17 @@ func Dial(rawAddr string) (int, error) {
 	opt := addr.Query()
 	noDelay := intopt(opt, "nodelay", 1)
 	if err = syscall.SetsockoptInt(fd, syscall.IPPROTO_TCP, syscall.TCP_NODELAY, noDelay); err != nil {
-		syscall.Close(fd)
+		Close(fd)
```
Comment (achille-roussel) [inline internal/sockets/dial.go:16]: Changed to use the `Close` function declared in the package which helped when tracking down where fd 0 was being closed.

File: wasi.go [hunk 3]
```diff
@@ -218,6 +218,9 @@ func (t *FileTable[T]) FDClose(ctx context.Context, fd FD) Errno {
 	if errno != ESUCCESS {
 		return errno
 	}
+	// We capture the file before removing the table entry because f is a
+	// pointer into the table and gets erased when the descriptor is deleted.
+	file := f.file
```
Comment (achille-roussel) [inline wasi.go:223]: This was the bug.

File: tracer.go [hunk 4]
```diff
@@ -791,7 +791,7 @@ func (t *Tracer) printSubscription(s Subscription) {
 func (t *Tracer) printEvent(e Event) {
 	t.printf("{EventType:%s,UserData:%#x", e.EventType, e.UserData)
 	if e.Errno != 0 {
-		t.printf(",Errno:%s", e.Errno.Name())
+		t.printf(",Errno:%s}", e.Errno.Name())
```
Comment (achille-roussel) [inline tracer.go:794]: Unrelated to the bug, I just fixed the formatting here.

**Review by chriso — APPROVED** [review]

**Pull request merged**